routes/course.js

```diff
@@ -262,4 +262,83 @@ router.delete("/api/courses/:course_id", auth, async (req, res) => {
   }
 });
 
+
+
+// delete and create a new course
+// need to provide new course as json object in request body
+// distribution field is also updated
+router.patch("/api/courses/:course_id", auth, async (req, res) => {
+  // delete the course
+  const c_id = req.params.course_id;
+  try {
+    // verify that course belongs to req user
+    const course = await Courses.findById(c_id).exec();
+    if (!course) {
+      return errorHandler(res, 404, "Course not found."); 
+    } else if (req.user._id !== course.user_id) {
+      return forbiddenHandler(res);
+    }
+    // delete course and update distributions
+    await Courses.findByIdAndDelete(c_id).exec();
+    await Years
+      .findByIdAndUpdate(
+        course.year_id,
+        { $pull: { courses: course._id }},
+        { runValidators: true }
+      )
+      .exec();
+    for (let id of course.distribution_ids) {
+      const distribution = await Distributions
+        .findByIdAndUpdate(
+          id,
+          { $pull: { courses: c_id } },
+          { new: true, runValidators: true }
+        )
+        .exec();
+      await distributionCreditUpdate(distribution, course, false);
+    }
+    returnData(course, res);
+  } catch (err) {
+    errorHandler(res, 500, err);
+  }
+
+  // insert a new course
+  const course = req.body.new_course;
```

Comment:
i'm not sure how update course works but does frontend send the new course body? is there a corresponding PR in the frontend that now uses this new route?

Reply:
Currently for posting a course, a new course body needs to be provided. So I think if we were to combine delete and add, we will need a new course as the request body. Do you think there might be better ways to do it?